packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -0,0 +1,341 @@
+    }}"
+    @mouseout="${x => {
+        x.isValidContentAndHasOverflow = false;
+    }}"
+    title=${x => (x.isValidContentAndHasOverflow ? x.label : null)}>
+    ${x => x.label}
+</span>`;
+```
+
+### Grouping
+
+Grouping will be based on the record value. The grouping header will display the rendered icon/spinner/text. In the case of an icon/spinner, it will also be followed by the `label` text. This will disambiguate cases where multiple record values map to the same icon (assuming the labels are different).
+
+For values that do not match any mapping, we will display the raw data value. While this introduces inconsistency, it seems preferable to the alternative, which is having multiple, separate groupings with a blank header (well, with just the item count in parens). Even in the case where there is a default mapping, we would still end up with separate groups with the identical default mapped icon and/or text, which is just as bad.
+
+Text in a grouping header will be ellipsized and gain a tooltip if there is not enough room to display it all.
+
+### Sorting
+
+Sorting will be based on the record value. For boolean and number values, a basic sort (just using basic comparison/equality operators) is the clear choice. For string values, it is less clear. In the case where the strings are enum values, they are likely to be non-localized, English strings. Using a basic (alphabetical) sort is not unreasonable. However, if there is a semantically meaningful sort order (e.g. "NOT_STARTED" < "RUNNING" < "DONE"), it would be nice to sort by that. We can only infer that semantic order from the order in which the mappings are given, e.g.:
+
+```HTML
+    <nimble-mapping-text key="NOT_STARTED" label="Not Started"></nimble-mapping-text>
+    <nimble-mapping-text key="RUNNING" label="Running"></nimble-mapping-text>
+    <nimble-mapping-text key="DONE" label="Done"></nimble-mapping-text>
+```
+
+We would need new support for sorting this way. We could define a new sorting option, "enumerated sort", where the column provides an ordered list of values, and the table sorts the column based on that given order. To compare the relative order of two values, we at have to search the list, making the sort operation a bit more expensive, but still probably reasonable except in the case of enums with many values.
+
+Our options are to use a basic sort (for all three value types), use an enumerated sort (again, for all three value types), or to start with a basic sort and implement the enumerated sort in a follow-on PR. Settling for a basic sort is reasonable and the easiest solution, but it would not give as nice an experience as an enumerated sort. The enumerated sort would expand the scope of this feature and could easily be implemented as a standalone change. For those reasons, I propose we use a basic sort for the initial submission of the column type, but we have a follow-on PR to add the enumerated sort and adopt that for the mapping/icon column.
```

Comment:
This is a bit nit-picky, but I don't think the terminology of saying something would be done in a "follow-on PR" is quite right. Since this is an HLD, its goal isn't to capture how many PRs will be involved in completing this work. I think the more accurate thing would be to say something like "this could be done as follow-on work that is out of scope for the initial implementation of these components".

Reply:
I've changed the wording a bit so I don't reference "PRs."